cpp/src/arrow/compute/kernels/isin-test.cc

```diff
@@ -0,0 +1,410 @@
+  std::vector<bool> expected;
+  char buf[20] = "test";
+
+  for (int32_t i = 0; i < kTotalValues * kRepeats; i++) {
+    int32_t index = i % kTotalValues;
+
+    ASSERT_GE(snprintf(buf + 4, sizeof(buf) - 4, "%d", index), 0);
+    values.emplace_back(buf);
+    member_set.emplace_back(buf);
+    expected.push_back(true);
+  }
+
+  CheckIsIn<BinaryType, std::string>(&this->ctx_, binary(), values, {}, member_set, {},
+                                     expected, {});
+
+  CheckIsIn<StringType, std::string>(&this->ctx_, utf8(), values, {}, member_set, {},
+                                     expected, {});
+}
+
+TEST_F(TestIsInKernel, IsInFixedSizeBinary) {
+  CheckIsIn<FixedSizeBinaryType, std::string>(
+      &this->ctx_, fixed_size_binary(5), {"bbbbb", "", "bbbbb", "aaaaa", "ccccc"},
+      {true, false, true, true, true}, {"bbbbb", "", "bbbbb", "aaaaa", "ccccc"},
+      {true, false, true, true, true}, {true, true, true, true, true}, {});
+
+  // Nulls in left
+  CheckIsIn<FixedSizeBinaryType, std::string>(
+      &this->ctx_, fixed_size_binary(5), {"bbbbb", "", "bbbbb", "aaaaa", "ccccc"},
+      {false, false, false, false, false}, {"bbbbb", "bbbbb", "aaaaa", "ccccc"},
+      {true, true, true, true, true}, {false, false, false, false, false}, {});
+
+  // Nulls in right
+  CheckIsIn<FixedSizeBinaryType, std::string>(
+      &this->ctx_, fixed_size_binary(5), {"bbbbb", "", "bbbbb", "aaaaa", "ccccc"},
+      {true, false, true, true, true}, {"bbbbb", "", "bbbbb", "aaaaa", "ccccc"},
+      {false, false, false, false, false}, {false, true, false, false, false}, {});
+
+  // Both array have Nulls
+  CheckIsIn<FixedSizeBinaryType, std::string>(
+      &this->ctx_, fixed_size_binary(5), {"bbbbb", "", "bbbbb", "aaaaa", "ccccc"},
+      {false, false, false, false, false}, {"bbbbb", "", "bbbbb", "aaaaa", "ccccc"},
+      {false, false, false, false, false}, {true, true, true, true, true}, {});
+
+  // No match
+  CheckIsIn<FixedSizeBinaryType, std::string>(
+      &this->ctx_, fixed_size_binary(5), {"bbbbc", "bbbbc", "aaaad", "cccca"},
+      {true, true, true, true}, {"bbbbb", "", "bbbbb", "aaaaa", "ccccc"},
+      {true, false, true, true, true}, {false, false, false, false}, {});
+
+  // Empty left array
+  CheckIsIn<FixedSizeBinaryType, std::string>(&this->ctx_, fixed_size_binary(5), {}, {},
+                                              {"bbbbb", "", "bbbbb", "aaaaa", "ccccc"},
+                                              {true, false, true, true, true}, {}, {});
+
+  // Empty right array
+  CheckIsIn<FixedSizeBinaryType, std::string>(
+      &this->ctx_, fixed_size_binary(5), {"bbbbb", "", "bbbbb", "aaaaa", "ccccc"},
+      {true, false, true, true, true}, {}, {}, {false, false, false, false, false}, {});
+
+  // Empty arrays
+  CheckIsIn<FixedSizeBinaryType, std::string>(&this->ctx_, fixed_size_binary(0), {}, {},
+                                              {}, {}, {}, {});
+}
+
+TEST_F(TestIsInKernel, FixedSizeBinaryResizeTable) {
```

Comment:
What does this test differently then above?

Reply:
Similar test in hash-test.cc. I'll remove this.